cmd/ova-provider-server/nfs_test.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/onsi/gomega"
+)
+
+func TestFindOVAFiles(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	tests := []struct {
+		name         string
+		setup        func(directory string)
+		expectedOVAs []string
+		expectedOVFs []string
+		expectError  bool
+	}{
+		{
+			name: "basic structure",
+			setup: func(directory string) {
+				ioutil.WriteFile(filepath.Join(directory, "test.ova"), []byte{}, 0644)
+				ioutil.WriteFile(filepath.Join(directory, "test.ovf"), []byte{}, 0644)
+				os.MkdirAll(filepath.Join(directory, "subdir1"), 0755)
```

Comment:
```suggestion
				os.MkdirAll(filepath.Join(directory, "subdir1", "subdir2"), 0755)
```
and I think it will be more structured if we place the creation of the directory structure at the beginning - for your consideration